wordpress-mobile/WordPress-Android
# Dashboard Personalization: Implement No Card State

## Closes 
#18956 

## Description
This PR implements the message when there is no cards in dashboard

## 📹 Screenshots
<img width="431" alt="Screenshot 2023-09-18 at 12 15 15 PM" src="https://github.com/wordpress-mobile/WordPress-Android/assets/17463767/73bee6e3-4cf1-4563-87fa-a99f85ea233a">

## To test:
1. Go to my site dashboard 
2. Go to Me → Debug Settings → enable `dashobard_personalization` flag
3. Go to My Site dashboard 
4. Click on **Personalize your home tab** card 
5. Hide all the cards
6. Verify that No cards message as shown in the screenshot above is shown 
7. Unhide any card
8. Verfiy that No cards message is not shown 

## Regression Notes
1. Potential unintended areas of impact
No cards message is not shown

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing and unit tests

3. What automated tests I added (or what prevented me from doing so)
Unit tests

PR submission checklist:

UI Changes testing checklist:
